### DISCUSSION TITLE
[Custom construtor] How to bind a cumstom construtor which returns a shared_ptr?

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#288] [Q&A] [answered]
In the porting guide, custom constructors. It says that o turn an existing factory function into a constructor, you will need to combine the above pattern with an invocation of the move/copy-constructor.
But what if the existing function returns shared_ptr and the move/copy-constructor are deleted so that I don't want invocation of the move/copy-constructor.

**Comment by wjakob** [comment, marked as answer]
You might be able to cook something up by delving into the details of how `stl/shared_ptr.h` is implemented. But this is not a supported feature, you will need to expose the factory as a method and not a constructor.

**Reply by huangweiwu** [reply]
Expose the factory as a method and not a constructor, this seems a good way to solve my problem.
Thanks a lot.
